tools/gfx/render.h

```diff
@@ -1113,37 +1342,36 @@ class IRenderer: public ISlangUnknown
     }
 
 // ----------------------------------------------------------------------------------------
-inline void IRenderer::setVertexBuffer(UInt slot, BufferResource* buffer, UInt stride, UInt offset)
+inline void IRenderer::setVertexBuffer(UInt slot, IBufferResource* buffer, UInt stride, UInt offset)
 {
     setVertexBuffers(slot, 1, &buffer, &stride, &offset);
 }
 
-/// Functions that are around Renderer and it's types
-struct RendererUtil
+// Global public functions
+
+extern "C"
 {
-    typedef SlangResult (*CreateFunc)(IRenderer** outRenderer);
+    typedef SlangResult(SLANG_MCALL * SGRendererCreateFunc)(IRenderer** outRenderer);
 
-        /// Gets the size in bytes of a Format type. Returns 0 if a size is not defined/invalid
-    SLANG_FORCE_INLINE static size_t getFormatSize(Format format) { return s_formatSize[int(format)]; }
-        /// Given a renderer type, gets a projection style
-    static ProjectionStyle getProjectionStyle(RendererType type);
+    /// Gets the size in bytes of a Format type. Returns 0 if a size is not defined/invalid
+    SLANG_GFX_API size_t SLANG_MCALL gfxGetFormatSize(Format format);
 
-        /// Given the projection style returns an 'identity' matrix, which ensures x,y mapping to pixels is the same on all targets
-    static void getIdentityProjection(ProjectionStyle style, float projMatrix[16]);
+    /// Gets the binding style from the type
+    SLANG_GFX_API BindingStyle SLANG_MCALL gfxGetBindingStyle(RendererType type);
 
-        /// Get the binding style from the type
-    static BindingStyle getBindingStyle(RendererType type) { return s_rendererTypeToBindingStyle[int(type)]; }
+    /// Given a renderer type, gets a projection style
+    SLANG_GFX_API ProjectionStyle SLANG_MCALL gfxGetProjectionStyle(RendererType type);
 
-        /// Get as text
-    static Slang::UnownedStringSlice toText(RendererType type);
+    /// Given the projection style returns an 'identity' matrix, which ensures x,y mapping to pixels
+    /// is the same on all targets
+    SLANG_GFX_API void SLANG_MCALL
+        gfxGetIdentityProjection(ProjectionStyle style, float projMatrix[16]);
 
-        /// Given a type returns a function that can construct it, or nullptr if there isn't one
-    static CreateFunc getCreateFunc(RendererType type);
+    /// Get the name of the renderer
+    SLANG_GFX_API const char* SLANG_MCALL gfxGetRendererName(RendererType type);
 
-    private:
-    static void compileTimeAsserts();
-    static const uint8_t s_formatSize[]; // Maps Format::XXX to a size in bytes;
-    static const BindingStyle s_rendererTypeToBindingStyle[];           ///< Maps a RendererType to a BindingStyle
-};
+    /// Given a type returns a function that can construct it, or nullptr if there isn't one
+    SLANG_GFX_API SGRendererCreateFunc SLANG_MCALL gfxGetCreateFunc(RendererType type);
+}
```

Comment:
It seems like we eventually want a more nuanced way to enumerate and find implementations of the graphics layer, but this works for now.

One thing I kind of have to wonder is: why have a function that returns a function pointer, that you then have to call? Why not just have a function that creates/returns a *thing* that you can work with?

Reply:
This is just what it is now, and this PR doesn't try to change anything meaningfully.

Reply:
I agree with the goals, and none of this nit-picks need to be addressed in this PR (the added `.zip` file does, but I see you've fixed that as I'm typing).